src/Features/CSharp/Portable/PopulateSwitch/CSharpPopulateSwitchCodeFixProvider.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes.PopulateSwitch
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.PopulateSwitch), Shared]
+    [ExtensionOrder(After = PredefinedCodeFixProviderNames.ImplementInterface)]
+    internal class CSharpPopulateSwitchCodeFixProvider : AbstractPopulateSwitchCodeFixProvider
+    {
+        protected override SyntaxNode GetSwitchStatementNode(SyntaxNode root, TextSpan span)
+        {
+            var token = root.FindToken(span.Start);
+            if (!token.Span.IntersectsWith(span))
+            {
+                return null;
+            }
+            
+            var switchExpression = (ExpressionSyntax)root.FindNode(span);
+            return (SwitchStatementSyntax) switchExpression.Parent;
+        }
+
+        protected override SyntaxNode GetSwitchExpression(SyntaxNode node)
+        {
+            var switchStatement = (SwitchStatementSyntax)node;
+            return switchStatement.Expression;
+        }
+
+        protected override List<string> GetMissingLabels(SyntaxNode node, SemanticModel model, INamedTypeSymbol enumType, out bool containsDefaultLabel)
+        {
+            var caseLabels = GetCaseLabels((SwitchStatementSyntax)node, out containsDefaultLabel);
+
+            var symbols = new List<ISymbol>();
+            foreach (var label in caseLabels)
+            {
+                // these are the labels like `MyEnum.EnumMember`
+                var memberAccessExpression = label as MemberAccessExpressionSyntax;
+                if (memberAccessExpression != null)
+                {
+                    var symbol = model.GetSymbolInfo(memberAccessExpression).Symbol;
+                    if (symbol != null)
+                    {
+                        symbols.Add(symbol);
+                    }
+                    continue;
+                }
+
+                // these are the labels like `EnumMember` (such as when using `using static Namespace.MyEnum;`)
+                var identifierName = label as IdentifierNameSyntax;
+                if (identifierName != null)
+                {
+                    var symbol = model.GetSymbolInfo(identifierName).Symbol;
+                    if (symbol != null)
+                    {
+                        symbols.Add(symbol);
+                    }
+                }
+            }
+
+            var missingLabels = new List<string>();
```

Comment:
Linq is allowed in the code fixes?  I can clean them up a lot with R#'s suggestions if they are.